float32 area
float32 xOffset
int8 detected